objectherkenning_openbare_ruimte/performance_evaluation_pipeline/source/oor_evaluation.py

```diff
@@ -290,101 +369,230 @@ def evaluate_coco(self) -> Dict[str, Dict[str, Dict[str, float]]]:
+            for target_cls_name, target_cls in coco_eval_classes.items():
                 logger.info(
                     f"Running custom COCO evaluation for {self.model_name} / {split if split != '' else 'all'} / {target_cls_name}"
                 )
                 eval = run_custom_coco_eval(
-                    coco_ground_truth_json=gt_json,
-                    coco_predictions_json=pred_json,
+                    coco_ground_truth_json=gt_json_files[i],
+                    coco_predictions_json=pred_json_files[i],
                     predicted_img_shape=self.ground_truth_image_shape,
                     classes=target_cls,
                     print_summary=False,
                 )
                 subkey = target_cls_name
                 custom_coco_result[key][subkey] = eval
+
+        # Remove temporary JSON files
+        for gt_file, pred_file in zip(gt_json_files, pred_json_files):
+            os.remove(gt_file)
+            os.remove(pred_file)
+
         return custom_coco_result
 
+    def save_tba_results_to_csv(self, results: Dict[str, Dict[str, Dict[str, float]]]):
+        """Save TBA results dict as CSV file."""
+        filename = os.path.join(self.output_folder, f"{self.model_name}-tba-eval.csv")
+        _df_to_csv(tba_result_to_df(results), filename)
 
-def tba_result_to_df(results: Dict[str, Dict[str, Dict[str, float]]]) -> pd.DataFrame:
-    """
-    Convert TBA results dictionary to Pandas DataFrame.
-    """
+    def save_per_image_results_to_csv(
+        self, results: Dict[str, Dict[str, Dict[str, float]]]
+    ):
+        """Save per-image results dict as CSV file."""
+        filename = os.path.join(
+            self.output_folder, f"{self.model_name}-per-image-eval.csv"
+        )
+        _df_to_csv(per_image_result_to_df(results), filename)
 
-    def _cat_to_header(cat: str) -> str:
-        """For nicer column headings we transform 'person_small' -> 'Person Small' etc."""
-        parts = [p.capitalize().replace("All", "ALL") for p in cat.split(sep="_")]
-        return " ".join(parts)
+    def save_coco_results_to_csv(self, results: Dict[str, Dict[str, Dict[str, float]]]):
+        """Save COCO results dict as CSV file."""
+        filename = os.path.join(
+            self.output_folder, f"{self.model_name}-custom-coco-eval.csv"
+        )
+        _df_to_csv(custom_coco_result_to_df(results), filename)
 
-    models = list(results.keys())
-    categories = list(results[models[0]].keys())
-    header = ["Model", "Split"]
-    header.extend([_cat_to_header(cat) for cat in categories])
+    def _compute_pr_curve_data(self, eval_func: Callable) -> pd.DataFrame:
+        """
+        Compute data needed to plot precision and recall curves. This method
+        calls either `self.evaluate_tba` or `self.evaluate_per_image` with
+        argument `single_size_only=True` for a range of confidence thresholds
+        and returns the results in a DataFrame.
```

Comment:
The docstrings misses that the function also calculates F1, F0.5, and F2 scores for various confidence thresholds. I would also change the name of the function to _compute_pr_f_curve_data to make it more readable.